alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/processor/model/Cycle.java

```diff
@@ -0,0 +1,36 @@
+package eu.xenit.alfresco.processor.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+/**
+ * Information object containing the parameters for the tracking activity.
+ */
+@Data
+@AllArgsConstructor
+public class Cycle {
```

Comment:
Are the txnId tracking and the commit time tracking used together? Or is it one or the other?

Reply:
Both: We are looking for X amount of transactions (batch size) starting from the First transaction, until specific time (txn start + txn increment)

Reply:
I'm asking because we have 2 sets of parallel concepts here.
- Start
- Step size
- current
One set for the id and one set for time

We could have separate Strategies that implement an interface that builds on the abstraction I mentioned

Reply:
I sent an invite to discuss these scenarios tomorrow morning

Reply:
ok! Enjoy your evening!